ai-services/internal/pkg/utils/retry.go

```diff
@@ -0,0 +1,55 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/project-ai-services/ai-services/internal/pkg/logger"
+)
+
+// BackoffFunc type definition
+type BackoffFunc func(currentDelay time.Duration) time.Duration
+
+// Retry -> retries based on the retry attempts and initialDelay time set on failure.
+// Does exponentialBackOff based on the provided BackoffFunc
+// Set backoff func to nil, if exponentialBackoff is not required
+func Retry(
+	attempts int,
+	initialDelay time.Duration,
+	backoff BackoffFunc,
+	fn func() error,
+) error {
+
+	delay := initialDelay
+	var err error
+
+	// Run the function initially and if no error do not proceed with retry attempts
+	err = fn()
+	if err == nil {
+		return nil
+	}
+
+	for i := range attempts {
+		logger.Infof("\n[Retry] Attempt %d/%d\n", i+1, attempts, 0)
```

Comment:
Does it print what is being retried like Configure/model download/application create ?